build_library/vm_image_util.sh

```diff
@@ -652,27 +639,38 @@ _write_qemu_conf() {
 
 _write_qemu_uefi_conf() {
     local script="$(_dst_dir)/$(_dst_name ".sh")"
-    local ovmf_ro="$(_dst_name "_ovmf_code.fd")"
-    local ovmf_rw="$(_dst_name "_ovmf_vars.fd")"
 
     _write_qemu_conf
-    cp "/usr/share/edk2-ovmf/OVMF_CODE.fd" "$(_dst_dir)/${ovmf_ro}"
-    cp "/usr/share/edk2-ovmf/OVMF_VARS.fd" "$(_dst_dir)/${ovmf_rw}"
-    sed -e "s%^VM_PFLASH_RO=.*%VM_PFLASH_RO='${ovmf_ro}'%" \
-        -e "s%^VM_PFLASH_RW=.*%VM_PFLASH_RW='${ovmf_rw}'%" -i "${script}"
-    VM_GENERATED_FILES+=( "$(_dst_dir)/${ovmf_ro}" "$(_dst_dir)/${ovmf_rw}" )
+
+    local flash_ro="$(_dst_name "_efi_code.fd")"
+    local flash_rw="$(_dst_name "_efi_vars.fd")"
+
+    case $BOARD in
+        amd64-usr)
+            cp "/usr/share/edk2-ovmf/OVMF_CODE.fd" "$(_dst_dir)/${flash_ro}"
+            cp "/usr/share/edk2-ovmf/OVMF_VARS.fd" "$(_dst_dir)/${flash_rw}"
+            ;;
+        arm64-usr)
+            # this bit of magic comes from http://tech.donghao.org/2014/12/18/running-fedora-21-on-qemu-system-aarch64/
+            cat "/build/${BOARD}/usr/share/edk2-armvirt/QEMU_EFI.fd" /dev/zero | dd iflag=fullblock bs=1M count=64 of="$(_dst_dir)/${flash_ro}"
+            dd if=/dev/zero bs=1M count=64 of="$(_dst_dir)/${flash_rw}"
+    esac
```

Comment:
For consistency in the file at least, we should put a ;; at the end of this last case.